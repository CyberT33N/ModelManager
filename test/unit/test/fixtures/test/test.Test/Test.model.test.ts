/*
███████████████████████████████████████████████████████████████████████████████
██******************** PRESENTED BY t33n Software ***************************██
██                                                                           ██
██                  ████████╗██████╗ ██████╗ ███╗   ██╗                      ██
██                  ╚══██╔══╝╚════██╗╚════██╗████╗  ██║                      ██
██                     ██║    █████╔╝ █████╔╝██╔██╗ ██║                      ██
██                     ██║    ╚═══██╗ ╚═══██╗██║╚██╗██║                      ██
██                     ██║   ██████╔╝██████╔╝██║ ╚████║                      ██
██                     ╚═╝   ╚═════╝ ╚═════╝ ╚═╝  ╚═══╝                      ██
██                                                                           ██
███████████████████████████████████████████████████████████████████████████████
███████████████████████████████████████████████████████████████████████████████
*/

// ==== DEPENDENCIES ====
import {
    describe, it, expect, beforeAll
} from 'vitest'

// ==== INTERNAL ====
import type { IFixtureDoc } from '@/src/FixturesManager'

describe('[UNIT TEST] - src/ModelUtils.ts', () => {
    let fixtureDoc: IFixtureDoc
     
    beforeAll(async() => {
        fixtureDoc = await import('@/test/fixtures/test/test.Test/0_test.ts') as IFixtureDoc
    })

    describe('[SNAPSHOT]', () => {
        it('should match the schema snapshot', () => {
            expect(fixtureDoc).toMatchSnapshot()
        })
    })
})