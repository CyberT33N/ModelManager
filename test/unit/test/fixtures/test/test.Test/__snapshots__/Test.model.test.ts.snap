// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[UNIT TEST] - src/ModelUtils.ts > [SNAPSHOT] > [INVALID] > [DUPLICATED] > should match the schema snapshot 1`] = `
{
  "docContents": {
    "__v": 0,
    "_id": "000000000000000000000003",
    "decimals": 18n,
    "name": "Unit Test fixture",
  },
  "name": "Example test fixture only for unit tests",
}
`;

exports[`[UNIT TEST] - src/ModelUtils.ts > [SNAPSHOT] > [INVALID] > [DUPLICATED] > should match the schema snapshot 2`] = `
{
  "docContents": {
    "__v": 0,
    "_id": "000000000000000000000003",
    "decimals": 18n,
    "name": "Unit Test fixture - other name but same id",
  },
  "name": "Example test fixture only for unit tests",
}
`;

exports[`[UNIT TEST] - src/ModelUtils.ts > [SNAPSHOT] > [VALID] > should match the schema snapshot 1`] = `
{
  "docContents": {
    "__v": 0,
    "_id": "000000000000000000000002",
    "decimals": 18n,
    "name": "Unit Test fixture",
  },
  "name": "Example test fixture only for unit tests",
}
`;

exports[`[UNIT TEST] - src/ModelUtils.ts > [SNAPSHOT] > [VALID] > should match the schema snapshot 2`] = `
{
  "docContents": {
    "__v": 0,
    "_id": "000000000000000000000004",
    "decimals": 18n,
    "name": "Unit Test fixture #2",
  },
  "name": "Example test fixture only for unit tests #2",
}
`;
